lib/record_store/provider/oracle_cloud_dns.rb

```diff
@@ -0,0 +1,182 @@
+            ttl: record.ttl,
+            rdata: record.rdata_txt,
+            operation: 'REMOVE',
+          ),
+        ]
+
+        client.patch_zone_records(
+          zone,
+          OCI::Dns::Models::PatchZoneRecordsDetails.new(items: patch_remove_record)
+        )
+      end
+
+      # Updates an existing record in the zone. It is expected this call modifies external state.
+      #
+      # Arguments:
+      # id - provider specific ID of record to update
+      # record - a kind of `Record` which the record with `id` should be updated to
+      def update(id, record, zone)
+        record_fqdn = Record.ensure_ends_without_dot(record.fqdn)
+
+        # Retrieve all records that you want to keep before it's updated because it will overwrite
+        all_records = []
+        client.get_zone_records(zone).each do |response|
+          all_records += response.data.items
+        end
+
+        update_zone_record_items = all_records.map do |r|
+          OCI::Dns::Models::RecordDetails.new(
+            domain: r.domain,
+            ttl: r.ttl,
+            rtype: r.rtype,
+            recordHash: r.record_hash,
+            rdata: r.rdata,
+            rrsetVersion: r.rrset_version,
+            isProtected: true,
+          )
+        end
+
+        update_zone_record_items << OCI::Dns::Models::RecordDetails.new(
+          domain: record_fqdn,
+          ttl: record.ttl,
+          rtype: record.type,
+          rdata: record.rdata_txt
+        )
+        update_zone_record_items.delete_if { |r| id == r.record_hash }
+
+        client.update_zone_records(
+          zone,
+          OCI::Dns::Models::UpdateZoneRecordsDetails.new(items: update_zone_record_items),
+        )
+      end
+
+      def secrets
+        super.fetch('oracle_cloud_dns')
+      end
+
+      def build_from_api(api_record)
+        fqdn = Record.ensure_ends_with_dot(api_record.domain)
+
+        record_type = api_record.rtype
+        return if record_type == 'SOA'
+
+        record = {
+          ttl: api_record.ttl,
+          fqdn: fqdn.downcase,
+          record_id: api_record.record_hash,
+        }
+        case record_type
+        when 'A', 'AAAA'
+          record[:address] = api_record.rdata
+        when 'ALIAS'
+          record[:alias] = api_record.rdata
+        when 'CAA'
+          flags, tag, value = api_record.rdata.split
+          record[:flags] = flags.to_i
+          record[:tag] = tag
+          record[:value] = Record.unquote(value)
+        when 'CNAME'
+          record[:cname] = api_record.rdata
+        when 'MX'
+          preference, exchange = api_record.rdata.split
+          record[:preference] = preference.to_i
+          record[:exchange] = exchange
+        when 'NS'
+          record[:nsdname] = api_record.rdata
+        when 'SPF', 'TXT'
+          record.merge!(txtdata: Record.unquote(api_record.rdata))
+        when 'SRV'
+          priority, weight, port, host = api_record.rdata.split(' ')
```

Comment:
Consider removing the argument to `split` as we can split on any whitespace.

```suggestion
          priority, weight, port, host = api_record.rdata.split
```